tools/get-contributor-list.sh

```diff
@@ -46,7 +40,7 @@ while IFS= read -r VER; do
 		read -r PREVIOUS_VERSION 
 		break 
 	fi 
-done < <( sed -n -E -e 's/^## \[?([0-9.]+)\]? - .*$/\1/p' "projects/$1/CHANGELOG.md" )
+done < <( sed -n -E -e 's/^## \[?([0-9]+.[0-9]+(.0)?)]? - .*$/\1/p' "projects/$1/CHANGELOG.md" )
```

Comment:
@anomiex Let me know if this regex is better for just pulling major versions, or if I've made a mess of things :)

Reply:
You need to escape the `.`s in the regex when they're not inside a character class, and you dropped a backslash before the `]`.
```suggestion
done < <( sed -n -E -e 's/^## \[?([0-9]+\.[0-9]+(\.0)?)\]? - .*$/\1/p' "projects/$1/CHANGELOG.md" )
```
Also this will consider "1.3.0" as the previous branch for Boost 1.3.2, even though Boost does in fact have a release branch for "1.3.1".

You might need to read the `.extra.changelogger.versioning` from composer.json (much like line 34) and use a different regex based on "wordpress" versus the default "semver". Or I suppose you could change lines 40–41 to something like
```sh
while IFS= read -r VER; do
    if git rev-parse --verify "origin/$PREFIX/branch-$VER" &>/dev/null; then
        PREVIOUS_VERSION="$VER"
        break 2
    fi
done
die "Could not find an existing branch for a version prior to $CURRENT_VERSION"
```
(while using the regex in my earlier comment) to keep looking until it finds a branch that exists.